Removed rule run for bulk content changes

orangejenny opened the pull request:
##### SUMMARY
https://dimagi-dev.atlassian.net/browse/ICDS-1206

Before merging
- [x] Update [docs](https://confluence.dimagi.com/display/commcarepublic/Bulk+download+and+upload+of+SMS+content+in+conditional+alerts)

##### PRODUCT DESCRIPTION
Prevents conditional alerts from re-processing when they're edited via the bulk upload UI.

orangejenny committed:
Removed rule run for bulk content changes

millerdev reviewed (commented):
I'm currently not remembering everything that can be changed via the bulk upload UI. Is it only message content?

orangejenny commented:
@millerdev Yep, only message content and the rule name.

orangejenny committed:
Fixed message dirty logic to not just look at last event

orangejenny committed:
Fixed bug in check_for_missing_content

At the point where this is called, messages may contain a mix of
translated and untranslated content. This is fine in the
TranslatedConditionalAlertUploader implementation, which just checked
for blank values, but the UntranslatedConditionalAlertUploader was
looking specifically for the key '*' and would break if it received a
translated message, which woould only have language code keys.

orangejenny committed:
Updated tests

orangejenny commented:
Updating tests turned up two unrelated, minor bugs. Hooray for tests.

orangejenny committed:
Fixed logic for checking for blank content

millerdev commented on corehq/messaging/scheduling/view_helpers.py:
```diff
@@ -233,9 +233,8 @@ def save_rule_messages(self, rule, rows):
-            missing = [lang for message in messages for value in message.values() if not value]
-            if missing:
-                raise RuleUpdateError(_("Missing content for {langs}").format(langs=", ".join(missing)))
+            if any([True for message in new_messages for value in message.values() if not value]):
```
nit
```suggestion
            if any(True for message in new_messages for value in message.values() if not value):
```

orangejenny committed:
Update corehq/messaging/scheduling/view_helpers.py

Co-Authored-By: Daniel Miller <dmiller@dimagi.com>

orangejenny committed:
Updated test to reflect change in error message

orangejenny committed:
Merge branch 'jls/skip-run-rule-on-bulk-upload' of github.com:dimagi/commcare-hq into jls/skip-run-rule-on-bulk-upload

orangejenny committed:
Fixed error messages in tests
